packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/code/CodeFile.java

```diff
@@ -15,29 +14,19 @@ public class CodeFile {
     private Dynamics dynamics;
 
     @TSIgnore
-    public void setProgrammingLanguageFromFileName(String fileName) {
+    public void setProgrammingLanguageFromFileName(final String fileName) {
         // Given the extension of a file, return the programming language
-        String[] parts = fileName.split("\\.");
-        String fileExtension = parts.length > 0 ? parts[parts.length - 1] : "";
-    
-        ProgrammingLanguage language = null;
-        switch (fileExtension) {
-            case "py":
-                language = ProgrammingLanguage.PYTHON;
-                break;
-            case "jl":
-                language = ProgrammingLanguage.Julia;
-                break;
-            case "r":
-                language = ProgrammingLanguage.R;
-                break;
-            case "zip":
-                language = ProgrammingLanguage.ZIP;
-                break;
-            default:
-                language = null; // TODO: Do we need an "unknown" language?
-        }
+        final String[] parts = fileName.split("\\.");
+        final String fileExtension = parts.length > 0 ? parts[parts.length - 1] : "";
+
+        final ProgrammingLanguage language = switch (fileExtension) {
+            case "py" -> ProgrammingLanguage.PYTHON;
+            case "jl" -> ProgrammingLanguage.Julia;
+            case "r" -> ProgrammingLanguage.R;
+            case "zip" -> ProgrammingLanguage.ZIP;
+            default -> null; // TODO: Do we need an "unknown" language?
```

Comment:
I'll say by default we could define them as `ProgrammingLanguage.TXT`?

Reply:
I like that... but im not going to do it here because if we're hitting this path `null` is working for now and i dont want to blow things up :D